pkg/github/github.go

```diff
@@ -596,3 +597,40 @@ func (g *GitHubClient) withRetries(ctx context.Context, retryFunc retry.RetryFun
 	}
 	return nil
 }
+
+type TokenSourceInputs struct {
+	GitHubToken string
+
+	GitHubAppID             string
+	GitHubAppPrivateKeyPEM  string
+	GitHubAppInstallationID string
+	GitHubRepo              string
+	Permissions             map[string]string
+}
+
+// TokenSource creates a token source from a GitHub token or GitHub App used for
+// authenticating a github client.
+func TokenSource(ctx context.Context, inputs *TokenSourceInputs) (githubauth.TokenSource, error) {
+	if inputs.GitHubToken != "" {
```

Comment:
You should check if inputs is nil and throw an error. This will panic otherwise when its nil.